service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NonConsultationResourceMapper.java

```diff
@@ -0,0 +1,180 @@
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String BLOOD_PRESSURE_CODE_2 = "<code code=\"109341000000100\" displayName=\"GP to GP communication transaction\""
+        + " codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String DIAGNOSTIC_REPORT_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication "
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String MEDICATION_REQUEST_CODE = "<code code=\"196391000000103\" displayName=\"Non-consultation medication "
+        + "data\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String CONDITION_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication transaction\" "
+        + "codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String QUESTIONNAIRE_RESPONSE_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication "
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String LOG_TEMPLATE = "Non-consultation resources mapped: {}";
+
+    private final MessageContext messageContext;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+    private final EncounterComponentsMapper encounterComponentsMapper;
+    private final Map<ResourceType, Function<String, EncounterTemplateParameters.EncounterTemplateParametersBuilder>> resourceBuilder =
+        Map.of(
+            ResourceType.Immunization, this::buildImmunization,
+            ResourceType.AllergyIntolerance, this::buildAllergyIntolerance,
+            ResourceType.ReferralRequest, this::buildReferralRequest,
+//          ResourceType.DiagnosticReport, this::buildDiagnosticRequest,
+            ResourceType.MedicationRequest, this::buildMedicationRequest,
+            ResourceType.Condition, this::buildCondition,
+            ResourceType.ProcedureRequest, this::buildProcedureRequest,
+            ResourceType.DocumentReference, this::buildDocumentReference);
+//          ResourceType.QuestionnaireResponse, this::buildQuestionnaireResponse);
+
+    public List<String> mapRemainingResourcesToEhrCompositions(Bundle bundle) {
+        var mappedResources = bundle.getEntry()
+            .stream()
+            .map(Bundle.BundleEntryComponent::getResource)
+            .filter(resource -> !messageContext.getIdMapper().hasIdBeenMapped(resource.getResourceType(), resource.getId()))
+            .filter(this::isMappableNonConsultationResource)
+            .map(this::mapResourceToEhrComposition)
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.toList());
+        LOGGER.debug(LOG_TEMPLATE, mappedResources.size());
+        return mappedResources;
+    }
+
+    private String mapResourceToEhrComposition(Resource resource) {
```

Comment:
If this may or may not give a result, it is better to use Optional<String>
Then inside it will be a bit easier to write the builder